File: BuyingTorches.cpp
#include<iostream>

using namespace std;

int main(){
    
    long long t, x, y, k;

    cin >> t;
    for(int i = 0; i < t; i++){
        cin >> x >> y >> k;
        long long sticksNeeded = y * k + k;
        long long sticks = 1;
        long long trades = 0;
        long long coal = 0;

        trades = sticksNeeded / (x - 1);
        trades += k / coal;
        
        if(sticksNeeded / x - 1 % 2 != 0){
            trades++;
        }
        
        

        if(k / coal % 2 != 0){
            trades++;
        }

        cout << trades << endl;
    }

}

File: BuyingTorches.java
import java.math.BigInteger;
import java.util.Scanner;

public class BuyingTorches {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int tests = sc.nextInt();

        for(int i = 0; i < tests; i++){
            int x = sc.nextInt();
            int y = sc.nextInt();
            int k = sc.nextInt();
            BigInteger trades = BigInteger.valueOf(0);
            BigInteger sticksNeeded = BigInteger.valueOf(y*k + k);
            BigInteger sticks = BigInteger.valueOf(1);
            int coal = 0;

            while (sticksNeeded.compareTo(sticks) > 0) {
                sticks = sticks.add(BigInteger.valueOf(x-1));
                trades = trades.add(BigInteger.valueOf(1));
                
            }
            while(coal < k){
                coal++;
                trades = trades.add(BigInteger.valueOf(1));
            }
            System.out.println(trades);
        }

    }
}
